### ISSUE TITLE
Length of weights not compatible with specified axis.

### ISSUE THREAD

**Issue by josepius-clemson** [MaartenGr/BERTopic#1054]
umap_model = UMAP(n_components=10, n_neighbors=50, min_dist=0.0)
print('umap done')
hdbscan_model = HDBSCAN(min_samples=15, gen_min_span_tree=True,prediction_data=True)
print('hdbscan done')

#vectorizer_model = CountVectorizer(stop_words="english")
ctfidf_model = ClassTfidfTransformer()
# representation_model = TextGeneration("gpt2")
representation_model = MaximalMarginalRelevance(diversity=0.6)
vectorizer_model = CountVectorizer(ngram_range=(2, 2), stop_words="english")

sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
pool = sentence_model.start_multi_process_pool()
embeddings = sentence_model.encode_multi_process(docs, pool)
print('embedding done')
# Pass the above models to be used in BERTopic
topic_model = BERTopic(embedding_model=sentence_model,
                       vectorizer_model=vectorizer_model,
                       umap_model=umap_model,
                       ctfidf_model=ctfidf_model,
                       hdbscan_model=hdbscan_model,
                       representation_model = representation_model,
                       nr_topics=15,
                      top_n_words = 15, n_gram_range=(1,2),low_memory = True,calculate_probabilities=True)

When I try to fit the model as follows, I am getting  following error:
topics, probs = topic_model.fit_transform(docs)
or 
topics, probs = topic_model.fit_transform(docs,embeddings)

      ValueError: Length of weights not compatible with specified axis.

Code works if I do not include representation_model . Pls advice how to fix this problem.

**Comment by MaartenGr** [comment]
I am not entirely sure but it may have something to do with `top_n_words`. Perhaps if you skip that parameter it may work as you also did not set the one in `MaximalMarginalRelevance`, so there might be a mismatch.

**Comment by josepius-clemson** [comment]
It worked. Thanks

**Comment by voxmenthe** [comment]
Yes as long as you set `top_n_words` the same in both your `MaximalMarginRelevance` model and your `BERTopic` model it should work.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this issue. Let me know if you want me to re-open the issue!
